addon/mixins/route.js

```diff
@@ -174,7 +174,8 @@ export default Ember.Mixin.create({
     }
 
     var params = Ember.merge(requestPayloadBase, options);
-    let promise = this.store[this._storeFindMethod](modelName, params);
+
+    let promise = this.afterInfinityModel(this.store[this._storeFindMethod](modelName, params));
```

Comment:
I think `afterInfinityModel` should be called on the resolved outcome of the promise - that way you know the object passed in is data, rather than a `thennable`.  

This is closer to how the regular `afterModel` works